modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/randomsql/RandomQuerySupplier.java

```diff
@@ -0,0 +1,246 @@
+
+        Ast cond;
+
+        if (rndWithRatio(4))
+            cond = new Const("TRUE");
+        else {
+            TableRef tbl = tbls0.get(rnd.nextInt(tbls0.size()));
+
+            cond = new BiCondition(
+                rndColumn(tbl),
+                new Const(Integer.toString(rnd.nextInt(100))),
+                Operator.EQUALS
+            );
+        }
+
+        // all tables connected one by one with an AND expression
+        while (tbls0.size() >= 2) {
+            TableRef left = tbls0.remove(0);
+            TableRef right = tbls0.get(0);
+
+            cond = new BiCondition(
+                cond,
+                new BiCondition(rndColumn(left), rndColumn(right), Operator.EQUALS),
+                Operator.AND
+            );
+        }
+
+        return cond;
+    }
+
+    /**
+     * Generates random SELECT expression.
+     *
+     * @param rndQryCtx Context of randomised query.
+     * @return Ast representing SELECT expression.
+     */
+    private Select rndSelect(RandomisedQueryContext rndQryCtx) {
+        Ast from = rndFrom(rndQryCtx);
+        Ast where = from instanceof TableList ? rndWhereClause(rndQryCtx.scopeTables()) : new Const("TRUE");
+
+        return new Select().from(from).where(where);
+    }
+
+    /**
+     * Generates random FROM expression.
+     *
+     * @param rndQryCtx Context of randomised query.
+     * @return Ast representing FROM expression.
+     */
+    private Ast rndFrom(RandomisedQueryContext rndQryCtx) {
+        List<TableRef> tbls = rndTables(rndQryCtx.schema());
+
+        assert !F.isEmpty(tbls);
+
+        if (tbls.size() == 1 || rndWithRatio(5)) {
+            tbls.forEach(rndQryCtx::addScopeTable);
+
+            return new TableList(tbls);
+        }
+
+        TableRef leftTbl = tbls.get(0);
+
+        Ast from = leftTbl;
+
+        rndQryCtx.addScopeTable(leftTbl);
+
+        for (int i = 1; i < tbls.size(); i++) {
+            TableRef rightTbl = tbls.get(i);
+
+            from = new InnerJoin(from, rightTbl,
+                new BiCondition(rndColumn(rndScopeTable(rndQryCtx)), rndColumn(rightTbl), Operator.EQUALS));
+
+            rndQryCtx.addScopeTable(rightTbl); // add a table to the scope AFTER we created the join condition
+        }
+
+        return from;
+    }
+
+    /**
+     * Returns random table which was used in FROM expression.
+     *
+     * @param rndQryCtx Context of randomised query.
+     */
+    private TableRef rndScopeTable(RandomisedQueryContext rndQryCtx) {
+        if (F.isEmpty(rndQryCtx.scopeTables()))
+            return null;
+
+        List<TableRef> tbls = new ArrayList<>(rndQryCtx.scopeTables());
+
+        if (tbls.size() > 1)
+            Collections.shuffle(tbls, rnd);
+
+        return tbls.get(0);
+    }
+
+    /**
+     * Returns random column from provided table.
+     *
+     * @param tbl Table.
+     * @return Random column.
+     */
+    private ColumnRef rndColumn(TableRef tbl) {
```

Comment:
Actually this return only "numeric" column.
Let's rename the method to be more specific.

Also, I'd suggest to add "string" column support (or may be some another varlen type) as well. 
I'm ok to add new types support within separate ticket.

Reply:
> Let's rename the method to be more specific

Ok 

> I'd suggest to add "string" column support

Actually this PR is like MVP for regression test. There are plans to improve generator not only with supporting different types, but with generating more meaningful queries, where conditions not totally random, but generated with regard to the value bounds and logic relations.